### PR TITLE
Allow zeroth order derivative for Bezier Curve

### PR DESCRIPTION
**Pull request by pepbos**
This PR changes `calcValue` to be a special case of `calcDerivative` with order zero.

This allows for less if / else switches further down the line and focuses calculations to a single function body.
Otherwise does not change the output of `calcValue` or `calcDerivative`.

### Brief summary of changes

In `SmoothSegmentedFunction`
- `calcDerivative` allows order zero.
- `calcValue` calls `calcDerivative` with order zero.

Similarly, in `SegmentedQuinticBezierToolkit`:
- `calcQuinticBezierCurveDerivU` and `calcQuinticBezierCurveDYDX` allow for order zero.
- `calcQuinticBezierCurveCurveVal` calls `calcQuinticBezierCurveDerivU` with order zero.

Changed `MustThrow` unit test accordingly.

### Testing I've completed

- Unit tests are still running.
- No significant change in performance (wall clock time) when simulating Rajagopal.
- No change in simulation result of Rajagopal.

### CHANGELOG.md

Change too minor?

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3596)

### CODE REVIEW

File: OpenSim/Common/SegmentedQuinticBezierToolkit.cpp [hunk 1]
```diff
@@ -362,15 +323,15 @@ double SegmentedQuinticBezierToolkit::calcQuinticBezierCurveDerivDYDX(
...
     //Bounds checking on the input
     SimTK_ERRCHK_ALWAYS( (u>=0 && u <= 1) , 
         "SegmentedQuinticBezierToolkit::calcQuinticBezierCurveDerivU", 
         "Error: double argument u must be between 0.0 and 1.0.");
 
-    SimTK_ERRCHK_ALWAYS( (order >= 1),
+    SimTK_ERRCHK_ALWAYS( (order >= 0),
         "SegmentedQuinticBezierToolkit::calcQuinticBezierCurveDerivU", 
-        "Error: order must be greater than.");
+        "Error: order must be greater than zero.");
```
Comment (nickbianco) [inline OpenSim/Common/SegmentedQuinticBezierToolkit.cpp:334]: Should be "...greater than or equal to zero."

**Review by nickbianco — CHANGES_REQUESTED** [review]
One minor comment, but otherwise looks good.

@pepbos I think one reviewer for this change is fine, max two.

**Comment by pepbos** [comment]
@nickbianco thanks for reviewing :)

**Review by nickbianco — APPROVED** [review]
LGTM.

**Pull request merged**